### PR TITLE
Fixes behemoth's earth pillar and AoE attacks not damaging vehicles outside of the middle

### PR DESCRIPTION
**Pull request by Wisemonster**
## About The Pull Request
Title, specifically them not doing damage to vehicles if they are tossed rather than rammed with landslide. Also makes it so the AoE attacks don't have to hit the center in order to do damage, rather they can hit any part of the tank.
## Why It's Good For The Game
Bugs bad
## Changelog
:cl:
fix: Behemoth's earth pillars should do damage to vehicles now if they are tossed at them
fix: Behemoth's AoE attacks will now damage all parts of the tank (previously only damaged it's center)
balance: Adjusted Behemoth's ability damage to compensate for attacks like meteor doing damage per vehicle tile hit
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm [hunk 1]
```diff
@@ -1366,6 +1366,8 @@
 	if(istype(hit_object, /obj/structure/reagent_dispensers/fueltank))
 		var/obj/structure/reagent_dispensers/fueltank/hit_tank = hit_object
 		hit_tank.explode()
+	if(istype(hit_object, /obj/hitbox)) // These don't hit the vehicles, but rather their hitboxes, so isarmored will not work here
+		return on_hit_anything(get_turf(hit_object), proj)
 	return rock_broke(get_turf(hit_object), proj)
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:1371]: I don't have a problem with what you're changing, but this whole behavior is just some weird ass stuff.
It does absolutely nothing if it hits an object (except specifically fuel tanks that just explode instantly?). Even if you make it hit tanks, it still excludes anything else like mechs at the very least? 
I assume not exploding when hitting other stuff is for walance, but if we want to hit tanks we should hit mechs as well.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm (hunk 1 above)
Comment (Wisemonster) [reply code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:1371]: That's weird, cause the only difference between the rock throw rocks and landslide rocks is landslide calls on_hit_anything always, while throwing them has this exception. And it seems that when a rock collides with the tank, it just teleports to the tanks middle to hit it, thus always hitting it's hitbox

Comment (Wisemonster) [reply code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:1371]: Or at the very least, it always hits the tank no matter what part it hits

**Comment by Wisemonster** [comment]
Alright, made it so that behemoth's area attack proc now actually checks for hitboxes when doing damage. This does mean that things like stomp and meteor will do damage based on how much of the vehicle is hit. To compensate, I reduced the damage the AoE attacks do so that meteor doesn't shred the tank (it does a lot, 200 damage, but it won't insta-kill the tank) This does mean that stomp will do less damage to the tank (though I'm not sure if behemoths even used it on the tank that much.)

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**